docker/build-and-upload.sh

```diff
@@ -1,23 +1,42 @@
 #!/bin/sh
-set -x
+# build script with multi-architecture support (linux/amd64 and linux/arm64)
+set -ex
+
 cd dockerfiles
-set -e
 tag=3007.6
-docker build -f dockerfile-saltmaster --tag erwindon/saltgui-saltmaster:$tag --tag erwindon/saltgui-saltmaster:latest .
-docker build -f dockerfile-saltmaster-tls --tag erwindon/saltgui-saltmaster-tls:$tag --tag erwindon/saltgui-saltmaster-tls:latest .
-docker build -f dockerfile-saltminion-ubuntu --tag erwindon/saltgui-saltminion-ubuntu:$tag --tag erwindon/saltgui-saltminion-ubuntu:latest .
-docker build -f dockerfile-saltminion-debian --tag erwindon/saltgui-saltminion-debian:$tag --tag erwindon/saltgui-saltminion-debian:latest .
-docker build -f dockerfile-saltminion-centos --tag erwindon/saltgui-saltminion-centos:$tag --tag erwindon/saltgui-saltminion-centos:latest .
+
+# Setup buildx for multi-architecture builds
+# Remove any existing problematic builder
+docker buildx rm multiarch 2>/dev/null || true
+# Create a fresh builder with multi-platform support
+docker buildx create --name multiarch --driver docker-container --use || true
+docker buildx inspect --bootstrap
+
+# Build function for multi-architecture images
+build_multiarch_image() {
+  dockerfile="$1"
+  imagename="$2"
+  
```

Comment:
can you please remove the whitespace on this otherwise empty line?
(feel free to use "push force")